### DISCUSSION TITLE
N-th git hook argument usage

### DISCUSSION THREAD

**Discussion by avudnez** [evilmartians/lefthook#597] [Q&A] [answered]
According to https://github.com/evilmartians/lefthook/blob/master/docs/configuration.md#run I should be able to have the following configuration to check commit messages:
```yml
commit-msg:
  skip:
    - merge
    - rebase
  parallel: true
  commands:
    check-commit-msg:
      run: python Git/hooks/commit-msg/check_commit_msg.py {1}
```
But the `{1}` is never expanded to the commit message file:
```shell
$ lefthook run commit-msg
Traceback (most recent call last):
  File "Git\hooks\commit-msg\check_commit_msg.py", line 104, in <module>
    exit(main())
  File "Git\hooks\commit-msg\check_commit_msg.py", line 82, in main
    with open(commit_message_file) as commit_message:
FileNotFoundError: [Errno 2] No such file or directory: '{1}'

exit status 1summary: (done in 0.06 seconds)
🥊  check-commit-msg
```

I don't see what I'm missing or doing differently from the docs. Can you help me ?

**Comment by mrexox** [comment, marked as answer]
Hey! I think you have this error because you run lefthook explicitly via `lefthook run commit-msg`. `{1}` will be a commit message file only when running via git hook, so in your case this will work:

```
lefthook install
git commit -m 'check my message, python script'
```

But if you want to test this hook, you can do the following:

```
echo 'check my message, python script' > /tmp/commit-message
lefthook run commit-msg /tmp/commit-message
```

I've appended an argument `/tmp/commit-message` to the explicit call of `lefthook`. In that case everything must work as expected,

**Reply by avudnez** [reply]
Of course, that makes sense 🙂 thanks !
